workspace-manager-clienttests/src/main/java/scripts/utils/ClientTestUtils.java

```diff
@@ -115,4 +130,40 @@ public static void assertHttpOk(WorkspaceApi workspaceApi, String label) {
         logger.debug("{} HTTP code: {}", label, httpCode);
         assertThat(HttpStatusCodes.isSuccess(httpCode), equalTo(true));
     }
+
+    /**
+     * Return a globally unique data reference name starting with the constant prefix.
+     * This will include a UUID reformatted to meet the rules for data reference names
+     * (just replacing hyphens with underscores). This method is useful when creating references
+     * on the same workspace from multiple threads.
+     * @return
+     */
+    public static String getUniqueDataReferenceName() {
+        return DATA_REFERENCE_NAME_PREFIX +
+            UUID.randomUUID()
+                .toString()
+                .replace("-", "_");
+    }
+
+    public static DataRepoSnapshot getTestDataReopSnapshot() {
+      return new DataRepoSnapshot()
+          .snapshot(TEST_SNAPSHOT)
+          .instanceName(TERRA_DATA_REPO_INSTANCE);
+    }
+
+    public static CreateDataReferenceRequestBody getTestCreateDataReferenceRequestBody() {
+      return new CreateDataReferenceRequestBody()
+          .name(getUniqueDataReferenceName())
+          .cloningInstructions(CloningInstructionsEnum.REFERENCE)
+          .referenceType(ReferenceTypeEnum.DATA_REPO_SNAPSHOT)
+          .reference(getTestDataReopSnapshot());
+    }
+
+    public static List<DataReferenceDescription> getDataReferenceDescriptions(
+        EnumerateDataReferences enumerateDataReferences, WorkspaceApi workspaceApi,
```

Comment:
Should this pass workspaceId directly instead of the entire EnumerateDataReferences test script object?

Reply:
Yes, it's a mis-refactor.